include/aws/mqtt/client.h

```diff
@@ -110,6 +110,40 @@ struct aws_mqtt_topic_subscription {
     void *on_publish_ud;
 };
 
+/**
+ * host_name                 The server name to connect to. This resource may be freed immediately on return.
+ * port                      The port on the server to connect to
+ * client_id                 The clientid to place in the CONNECT packet.
+ * socket_options            The socket options to pass to the aws_client_bootstrap functions.
+ *                           This is copied into the connection
+ * tls_options               TLS settings to use when opening a connection.
+ *                           This is copied into the connection
+ *                           Pass NULL to connect without TLS (NOT RECOMMENDED)
+ * clean_session             True to discard all server session data and start fresh
+ * keep_alive_time_secs      The keep alive value to place in the CONNECT PACKET, a PING will automatically
+ *                           be sent at this interval as well. If you specify 0, defaults will be used
+ *                           and a ping will be sent every 60s. If you're tempted to set this really low
+ *                           to find out about connection drops quickly, this is going to consume a lot of cpu.
+ *                           Instead, consider setting tcp keep alive to low timeout values and leaving the ping
```

Comment:
after our conversation, I think it's worth keeping the comments that tcp-timeout might be a lower-power way to accomplish this, but might not work on every platform